pkgs/applications/graphics/inkscape/extensions/silhouette/default.nix

```diff
@@ -0,0 +1,78 @@
+{ fetchFromGitHub
+, lib
+, pkgs
+, python3
+, umockdev
+, writeScript
+}:
+
+let
+  launch-sendto_silhouette = writeScript "sendto_silhouette.sh" ''
+    cd $(dirname $0)
+    ./sendto_silhouette.py "$@"
+  '';
+  launch-silhouette_multi = writeScript "silhouette_multi.sh" ''
+    cd $(dirname $0)
+    ./silhouette_multi.py "$@"
+  '';
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "inkscape-silhouette";
+  version = "1.28.pre+unstable=2023-06-03"; # Note: there has been lot of as-yet-not-released development since 1.27 was released: https://github.com/fablabnbg/inkscape-silhouette/releases/tag/v1.27
+
+  src = fetchFromGitHub {
+    owner = "fablabnbg";
+    repo = pname;
+    rev = "a1941b6f740466a20dfa116415dd599b1e496302";
+    sha256 = "sha256-CLmSHIhaqtSzPXB1HgHPG+fRW7ktcN3Y3IEWuCVLarw=";
+  };
+
+  patches = [
+    ./interpreter.patch
+  ];
+
+  propagatedBuildInputs = [
+    python3.pkgs.pyusb
+    python3.pkgs.lxml
+    python3.pkgs.inkex
+    python3.pkgs.matplotlib
+  ];
+
+  nativeCheckInputs = [
+    python3.pkgs.pytestCheckHook
+    umockdev
+  ];
+
+  pytestFlagsArray = [
+    "test"
+  ];
+
+  format = "setuptools";
+
+  doCheck = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/inkscape/extensions
+    make install DESTDIR=$out PREFIX=
+    # Unmark read_dump.py as executable so wrapPythonProgramsIn won't turn it into a shell script (thereby making it impossible to import as a Python module).
+    chmod -x $out/share/inkscape/extensions/silhouette/read_dump.py
+    cp ${launch-sendto_silhouette} $out/share/inkscape/extensions/sendto_silhouette.sh
+    cp ${launch-silhouette_multi} $out/share/inkscape/extensions/silhouette_multi.sh
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    wrapPythonProgramsIn "$out/share/inkscape/extensions/" "$out $pythonPath"
+  '';
+
+  meta = with lib; {
+    description = "An extension to drive Silhouette vinyl cutters (e.g. Cameo, Portrait, Curio series) from within inkscape.";
+    homepage = "https://github.com/fablabnbg/inkscape-silhouette";
+    license = licenses.gpl2;
```

Comment:
`gpl2` is unclear, use `gpl2Plus` or `gpl2Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

Reply:
Oops, I'm not sure how that slipped through the cracks, sorry! Changed to `gpl2Plus`.

Reply:
How did you determine the `Plus`? I don't see any "or later" in https://github.com/search?q=repo%3Afablabnbg%2Finkscape-silhouette%20GPL&type=code

Reply:
I don't know much about software licenses, but I got this from here https://github.com/fablabnbg/inkscape-silhouette/blob/a1941b6f740466a20dfa116415dd599b1e496302/LICENSE#L298-L299 (emphasis mine):

> either version 2 of the License, or (at your option) any later version.

Reply:
That section is part of the "How to Apply These Terms to Your New Programs" documentation for the license, but I don't see the project actually _do_ that AFAICS

Reply:
Ahh, I see. Then I really have no idea what the license is for this software. It doesn't seem to follow the GPL advice to include a notice on each file. Per the instructions [here](https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206), I've reached out to the maintainer [here](https://github.com/fablabnbg/inkscape-silhouette/issues/239) to see clarity.

What does this mean for this PR? Do we have to wait until we have a clear answer on this, or can we use the ambiguous `gpl2` until we get an answer on that github issue I posted?

Reply:
The code is effectively GPLv2 only.

- setup.py:license='GPL-2.0',
- silhouette/Graphtec.py:# Distribute under GPLv2 or ask.
- misc/test_drive.py:# (c) 2013 jnweiger@gmail.com, distribute under GPL-2.0
- silhouette/convert2dashes.py: [GPL] either version 2 of the License, or (at your option) any later version.
- sendto_silhouette.py:# (C) 2013 jw@suse.de. Licensed under CC-BY-SA-3.0 or GPL-2.0 at your choice.
- silhouette-udev-notify.sh:# (c) 2013, jw@suse.de - All rights reserved. Distribute under GPL-2.0


While the main file sendto_silhouette.py is under a dual license we need to find what applies to the entire package. CC BY-SA-3.0 is not mentioned anywhere else, so we have at least 3 files that only allow GPLv2.

Regarding the hint in Graphtec.y: "Distribute under GPLv2 or ask." - I probably won't object to allow "GPLv2 or later", if someone asks for it.
But: by now the code base has contributions from many different authors, so my vote is insufficient.

I understand that the text of GPLv2 itself contains the suggestion to use it as "or later" https://github.com/fablabnbg/inkscape-silhouette/blob/main/LICENSE#L299
But this package did not follow that suggestion.

Reply:
Thank you for the clarification, @jnweiger!